tests/test_pipeline_run_script.py

```diff
@@ -1,32 +1,116 @@
+            {},
+            [],
+            id="script_pipeline_a0_b0_dependencies",
+        ),
+        pytest.param(
+            "housing",
+            "",
+            ["p value"],
+            "script_pipeline_housing_simple",
+            {},
+            {},
+            [],
+            id="script_pipeline_housing_simple",
+        ),
+        pytest.param(
+            "housing",
+            "",
+            ["y", "p value"],
+            "script_pipeline_housing_multiple",
+            {},
+            {},
+            [],
+            id="script_pipeline_housing_multiple",
+        ),
+        pytest.param(
+            "housing",
+            "",
+            ["y", "p value"],
+            "script_pipeline_housing_w_dependencies",
+            {"p value": {"y"}},
+            {},
+            [],
+            id="script_pipeline_housing_w_dependencies",
+        ),
+    ],
+)
 @pytest.mark.slow
-def test_run_script_dag(virtualenv, tmp_path):
+def test_run_script_dag(
+    virtualenv,
+    tmp_path,
+    linea_db,
+    execute,
+    input_script1,
+    input_script2,
+    artifact_list,
+    pipeline_name,
+    dependencies,
+    dag_config,
+    input_parameters,
+):
     """
     Verifies that the DAGs we produce do run successfully.
     """
 
-    # FIXME - we need a more basic test than this that does not depend on pandas and sklearn etc.
-    # installing them takes way too long right now.
-
-    # Copy the dag and the data
-    # NOTE: Don't want to leave them in the tests folder, since there are other
-    # files we need to copy without which the dag will fail.
-    subprocess.check_call(
-        [
-            "cp",
-            "-f",
-            "tests/unit/plugins/expected/script_pipeline_housing_simple/script_pipeline_housing_simple_module.py",
-            str(tmp_path),
-        ]
+    code1 = Path(
+        "tests", "unit", "graph_reader", "inputs", input_script1
+    ).read_text()
+    execute(code1, snapshot=False)
+
+    if input_script2 != "":
+        code2 = Path(
+            "tests", "unit", "graph_reader", "inputs", input_script2
+        ).read_text()
+        execute(code2, snapshot=False)
+
+    # Write out pipeline files
+    artifact_def_list = [get_lineaartifactdef(art) for art in artifact_list]
+    artifact_collection = ArtifactCollection(
+        linea_db, artifact_def_list, input_parameters=input_parameters
     )
 
+    # Construct pipeline writer
+    pipeline_writer = PipelineWriterFactory.get(
+        pipeline_type=PipelineType["SCRIPT"],
+        artifact_collection=artifact_collection,
+        dependencies=dependencies,
+        pipeline_name=pipeline_name,
+        output_dir=tmp_path,
+        dag_config=dag_config,
+    )
+    pipeline_writer.write_pipeline_files()
+
     os.chdir(str(tmp_path))
 
-    subprocess.check_call(
-        ["python", "script_pipeline_housing_simple_module.py"]
-    )
+    subprocess.check_call(["python", f"{pipeline_name}_module.py"])
```

Comment:
can you catch it if the subprocess actually fails? not sure if this replaces an assert statement.

Reply:
i realise we didnt have it before too. can you please add the exit status check?

Reply:
Check_call will raise an exception failing the test if the return code is non-zero. 
[https://docs.python.org/3/library/subprocess.html#exceptions](https://docs.python.org/3/library/subprocess.html#exceptions 
)
Do we still want to do an explicit check?